src/fiar_notify_handler.erl

```diff
@@ -33,19 +33,19 @@ init(_InitArgs, _LastEventId, Req) ->
         process_register(MatchId, UserId),
         {ok, Req2, #{user => User}};
       {not_authenticated, _AuthHeader, Req1} ->
-        {shutdown, 401, [], [], Req1, #{}}
+        {shutdown, 401, [], [], Req1}
     end
   catch
-    _:notfound -> 
-      {shutdown, 404, [], [], Req, #{}}
+    _:notfound ->
+      {shutdown, 404, [], [], Req}
   end.
 
 handle_notify({match_updated, Match}, State) ->
   MatchJson = jiffy:encode(fiar_match:to_json(Match)),
-  {send, [{data, MatchJson}, {name, <<"turn">>}], State};
+  {send, [{data, <<MatchJson>>}, {name, <<"turn">>}], State};
```

Comment:
I was trying to fix this dialyzer error:

< fiar_notify_handler.erl:43: The inferred return type of handle_notify/2
({'send',[{'data',<<_:8>>} | {'name',<<_:32,_:_*56>>},...],_}) has nothing
in common with {'nosend',_} | {'stop',_} | {'send',#{},_}, which is the
expected return type for the callback of lasse_handler behaviour

But after rerunning dialyzer both with and wthout the << >> , it looks lke
that doesn't make a difference


On Sun, Jul 19, 2015 at 3:01 PM, Brujo Benavides <notifications@github.com>
wrote:

> In src/fiar_notify_handler.erl
> <https://github.com/inaka/fiar/pull/52#discussion_r34960941>:
>
> >    end.
> >
> >  handle_notify({match_updated, Match}, State) ->
> >    MatchJson = jiffy:encode(fiar_match:to_json(Match)),
> > -  {send, [{data, MatchJson}, {name, <<"turn">>}], State};
> > +  {send, [{data, <<MatchJson>>}, {name, <<"turn">>}], State};
>
> This doesn't look right. Adding << to the sides of an iolist does *not*
> turn it into a binary. What are you trying to achieve here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inaka/fiar/pull/52/files#r34960941>.
>